### ISSUE TITLE
Fix crash when unserializing references in php 7.4 typed properties

### ISSUE THREAD

**Issue by TysonAndre** [igbinary/igbinary#205]
https://wiki.php.net/rfc/typed_properties_v2

PHP's own serializer in ext/standard is a good place to look at to see how unserialization will be affected (That PR doesn't seem to touch the serializer)

- How will unserializers handle unserializing references to properties while supporting that RFC?
- What error handling needs to be done? (When setting property values, when modifying a reference that includes a property value, etc.)
- Look into how the new type `uninitialized` (of a typed property) is represented (It seems like it is separate from unset or null)

**Comment by TysonAndre** [comment]
This was released 2 days ago.

The changes that were made to the standard serializer for PHP 7.4 (i.e. `serialize()`/`unserialize()` (and corresponding tests) can be seen with the following command:

```bash
cd /path/to/php-src
git show e219ec144ef6682b71e135fd18654ee1bb4676b4 ext/standard
```

Similar changes and tests will need to be added to igbinary for PHP 7.4 to work.

**Comment by TysonAndre** [comment]
```php
class A {
	public int $a;
	public $b;
}

$a = new A();
$a->a = 1234;
$a->b = &$a->a;
var_dump(bin2hex($s = igbinary_serialize($a)));
var_dump(igbinary_unserialize($s));  // this crashes
```

**Comment by Archanium** [comment]
Anything that can be done with helping this one along? :)

**Comment by TysonAndre** [comment]
That crash was fixed in #248 - I just forgot to close this issue.
